Create additional settings modules (#1490)
ansible/aap-docs: downstream/modules/platform/proc-settings-platform-gateway.adoc

[id="proc-settings-platform-gateway"]

= Platform gateway

//To be added to Donna's AAP/UI document for 2.5 

> The Platform gateway is the service that handles authentication and authorization for {PlatformNameShort}. 

Reviewer: ```suggestion
The Platform gateway is the service that handles authentication and authorization for {PlatformNameShort}. 
```
Do we have an attribute for Platform gateway?

Author: No, there's one for the menu selection.

It provides a single ingress into the Platform and serves the Platform's user interface.

Use the following procedure to configure the options.

.Procedure
. From the navigation panel, select {MenuSetGateway}.
. The Platform gateway settings page is displayed. 
+
image::platform_gateway_settings_page.png[Initial platform gateway settings page]
+
. To configure the options, click btn:[Edit].
+
image::platform_gateway_full.png[Platform gateway configurable options]
+

> . The following platform gateway options can be configured:

Reviewer: ```suggestion
. You can configure the following platform gateway options:
```

* *Platform gateway basic auth enabled*: Enable basic authentication to the platform gateway API.
+
Turning this off prevents all basic authentication (local users), so customers need to make sure they have their alternative authentication mechanisms correctly configured before doing so. 
+
Turning it off with only local authentication configured also prevents all access to the UI.
+
*Social auth username in full email*: Enabling this setting alerts social authentication to use the full email as username instead of the full name.
+
*Platform gateway token name*: The header name to push from the proxy to the backend service. 
+
[WARNING]
==== 
If this name is changed, backends must be updated to compensate.
====
+
* *Platform gateway access token expiration*: How long the access tokens are valid for.
* *Jwt private key*: TBD
* *Jwt public key*: (Read only) TBD

You can configure the following Session options:

* *Session cookie age*: Time in seconds before a session expires.

You can configure the following Password security options:

* *Password min upper*: How many uppercase characters need to be in a local password.

> * *Password min length*: How long does a local password have to be.

Reviewer: ```suggestion
* *Password min length*: The minimum length of a local password.
```

* *Password min digits*: How many numerical characters need to be in a local password.
* *Password min special*: How many special characters need to be in a local password.

You can configure the following Custom login options:

> * *Custom login info*: Provide a specific information (such as a legal notice or a disclaimer) to a text box in the login modal.

Reviewer: ```suggestion
* *Custom login info*: Provide specific information (such as a legal notice or a disclaimer) to a text box in the login modal.
```
Are we allowed to use modal?

* *Custom logo* : Provide an image file for setting up a custom logo (must be a data URL with a base64-encoded GIF, PNG, or JPEG image).
For more information, see xref:proc-custom-logos-images[Setting a custom logo].

You can configure the following additional settings:

> * *Jwt expiration buffer in seconds*: The number of seconds before a JWT token's expiration to revoke from the cache.

Reviewer: Should Jwt  be uppercase?

+
When authentication happens a JWT token is created for the user and that token is cached. 
When subsequent calls happen to services such as automation controller or EDA, the token is taken from the cache and sent to the service. 
Both the token and the cache of the token have an expiration time. 
If the token expires while in the cache the authentication process attempts results in a 401 error (unauthorized). 
This setting gives {PlatformName} a buffer by removing the JWT token from the cache before the token expires. 
When a token is revoked from cache a new token with a new expiration is generated and cached for the user. 
As a result, expired tokens from the cache are never used. 
This setting defaults to 2 seconds. 

> If you have a large latency between platform gateway and your services and observe 401 responses you might want to increase this setting to combat the 401 responses.

Reviewer: ```suggestion
If you have a large latency between platform gateway and your services and observe 401 responses you can increase this setting to lower the number of 401 responses.
```
